portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,68 @@
 
 package com.google.sps.servlets;
 
+import com.google.sps.servlets.Comment;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import java.util.List;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that handles comments data. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private List<Comment> comments = new ArrayList<Comment>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+
+    // Convert the comments into JSON.
+    String json = convertToJson(comments);
+
+    // Send the JSON as the response.
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get input from the form.
+    String text = getParameter(request, "comment-text", "");
+    String commenterName = getParameter(request, "commenter-name", "Anonymous");
+    String commenterEmail = getParameter(request, "commenter-email", "Unknown");
+
+    // Create new Comment object to store input.
+    Comment newComment = new Comment(text, commenterName, commenterEmail);
```

Comment:
Usually I would like to inline variables if it is only used once, but I understand that this line could become very complicate so it depends on you.

But I think you can definitely inline newComment.

Reply:
Done - will try to inline variables only used once.